src/main/java/in/koreatech/koin/domain/bus/controller/BusController.java

```diff
@@ -0,0 +1,27 @@
+package in.koreatech.koin.domain.bus.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import in.koreatech.koin.domain.bus.dto.BusRemainTimeResponse;
+import in.koreatech.koin.domain.bus.service.BusService;
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequiredArgsConstructor
+public class BusController implements BusApi {
+
+    private final BusService busService;
+
+    @GetMapping("/bus")
+    public ResponseEntity<BusRemainTimeResponse> getBusRemainTime(
+        @RequestParam(value = "bus_type") String busType,
+        @RequestParam String depart,
+        @RequestParam String arrival
```

Comment:
enum에 선언해둔 정적 팩토리 메소드로 선언해둔 string -> enum의 파싱 로직에다가 `@JsonCreator` 달아두면 여기에 Enum을 선언해서 바로 사용할 수 있을거같은데 한번 수정해보겠어요?
```suggestion
        @RequestParam(value = "bus_type") BusType busType,
        @RequestParam BusStation depart,
        @RequestParam BusStation arrival
```
잘 모르겠으면 INTERNAL 코드 한번 참고해보세용 (잘 모르겠으면 다희한테도 물어보세유 저번에 알려줬었는데)

Reply:
이런 기능이 있는 줄 몰랐네요..!! 정말 좋은 것 같습니다. 👍

다만 적용하는 과정에서 몇 가지 이슈가 발생했습니다.
1. `@RequestParam`을 사용하면 enum 매핑에 `@JsonCreator`를 활용할 수 없습니다.
    - 매핑 자체는 가능했습니다. 다만 `@JsonCreator`를 통해 생성 로직을 제어하기 위해서는 `@RequestBody`를 사용해야만 합니다.
    - `@RequestParam` 사용 시 `valueOf()`메서드를 사용하기에 로직을 임의로 수정할 수 없습니다.
    - 현재 enum 요소들의 네이밍은 UPPER_CASE로 통일되어 있는데, API 요청에 대해서도 대문자로 받아야 합니다.
        - 기존에는 소문자로 받았기에 하위호환성이 우려됩니다. (대문자로 안받으면 에러 발생)
2. `@RequestBody`로 변경하면 API 요청 구조가 달라집니다.
    - 기존에는 파라미터로 받던 것을 body로 받으니 하위호환성 이슈가 발생합니다.
3. `@RequestBody`로 변경하면 POST(PUT) 요청으로 변경해야 합니다.
    - 당연히 하위호환성 이슈가 발생합니다. (기존: GET)
    - 단순 조회 요청을 POST로 받는 것이 적절한지 의심됩니다.

결론을 내리자면 저는 개인적으로 이번 API에서 제안해주신 기능을 사용하지 않는 것이 적절하다고 생각합니다.
선배님의 의견은 어떤지 궁금합니다.

+) 혹시 내용 중에 잘못된 부분이 있다면 말씀해주시면 감사하겠습니다.

Reply:
오~! RequsetParam의 경우 valueof() 메소드를 사용하는군요
확실히 말한대로 이번 경우는 사용하지 않는 편이 좋은 것 같아요
알아보고 정리해줘서 고맙습니다 👍

Reply:
아시는 내용일 수도 있지만 간단하게 언급만 하겠습니다

Converter를 사용하면 `@RequestParam`을 유지하면서 자동으로 임의의 로직을 통해 형변환을 진행할 수 있습니다.
기존 코인에서 컨버터를 사용한 사례가 있었는데, 스프링 부트에서는 훨씬 쉽게 구현이 가능해서 마음에 들었습니다.
다만 enum은 covertor을 implement하는 것이 불가능하다 보니 Converter 클래스를 새로 작성해야 합니다.
결국 계층 하나가 더 생기는 것이기에 오히려 더 불편해질 수 있겠다는 생각이 듭니다.
지금은 이대로 유지하겠습니다.

참고) https://github.com/BCSDLab/KOIN_API/pull/297